src/server_manager/web_app/ui_components/app-root.js

```diff
@@ -526,17 +580,20 @@ export class AppRoot extends mixinBehaviors
 
   /**
    * Sets the language and direction for the application
-   * @param {string} newLanguage
-   * @param {string} langDir
+   * @param {string} language
+   * @param {string} direction
    */
-  setLanguage(newLanguage, langDir) {
-    const messagesUrl = `./messages/${newLanguage}.json`;
-    this.loadResources(messagesUrl, newLanguage);
-    if (langDir === 'rtl') {
-      this.$.appDrawer.align = 'right';
-      this.$.sideBar.align = 'right';
-    }
-    this.language = newLanguage;
+  setLanguage(language, direction) {
+    const messagesUrl = `./messages/${language}.json`;
+    this.loadResources(messagesUrl, language);
+
+    document.documentElement.setAttribute('dir', direction);
+    const alignDir = direction === 'ltr' ? 'left' : 'right';
+    this.$.appDrawer.align = alignDir;
+    this.$.sideBar.align = alignDir;
+
+    window.localStorage.setItem('overrideLanguage', language);
```

Comment:
Humm... It's better if the UI components don't know about localStorage. It ties the component to an application. It's a hidden global dependency.

Can you move this logic to app.ts?
Same for the document.documentElement attribute, since it's not part of the AppRoot element..

Reply:
Done